apisearch.php

```diff
@@ -19,120 +19,142 @@
+    {
+        Configuration::updateValue('AS_CLUSTER_URL', '');
+        Configuration::updateValue('AS_ADMIN_URL', Defaults::DEFAULT_AS_ADMIN_URL);
+        Configuration::updateValue('AS_API_VERSION', Defaults::DEFAULT_AS_API_VERSION);
+        Configuration::updateValue('AS_APP', '');
+        Configuration::updateValue('AS_INDEX', '');
+        Configuration::updateValue('AS_TOKEN', '');
+        Configuration::updateValue('AS_SHOP', '');
+        Configuration::updateValue('AS_INDEX_PRODUCTS_WITHOUT_IMAGE', Defaults::DEFAULT_INDEX_PRODUCTS_WITHOUT_IMAGE);
+        Configuration::updateValue('AS_REAL_TIME_INDEXATION', Defaults::DEFAULT_REAL_TIME_INDEXATION);
+
+        $meta_as = new Meta();
+        $meta_as->page = 'module-apisearch-as_search';
+        $meta_as->title = $this->l('Apisearch - Search');
+        $meta_as->description = $this->l('Search by Apisearch');
+        $meta_as->url_rewrite = $this->l('as_search');
+
+        if (!$meta_as->save()) {
+            return false;
+        }
 
-    if (!$meta_as->delete()) {
-      return false;
+        return parent::install() &&
+            $this->registerHook('header') &&
+            $this->registerHook('top') &&
+            $this->registerHook('actionObjectProductAddAfter') &&
+            $this->registerHook('actionObjectProductUpdateAfter') &&
+            $this->registerHook('actionObjectProductDeleteBefore') &&
+            $this->registerHook('actionObjectOrderUpdateAfter');
     }
 
-    return parent::uninstall();
-  }
+    public function uninstall()
+    {
+        Configuration::deleteByName('AS_CLUSTER_URL');
+        Configuration::deleteByName('AS_ADMIN_URL');
+        Configuration::deleteByName('AS_API_VERSION');
+        Configuration::deleteByName('AS_APP');
+        Configuration::deleteByName('AS_INDEX');
+        Configuration::deleteByName('AS_TOKEN');
+        Configuration::deleteByName('AS_SHOP');
+        Configuration::deleteByName('AS_INDEX_PRODUCTS_WITHOUT_IMAGE');
+        Configuration::deleteByName('AS_REAL_TIME_INDEXATION');
+
+        $meta_as = Meta::getMetaByPage('module-apisearch-as_search', Context::getContext()->language->id);
+        $meta_as = new Meta($meta_as['id_meta']);
+
+        if (!$meta_as->delete()) {
+            return false;
+        }
 
-  public function getContent() {
-    if (((bool) Tools::isSubmit('submitApisearchModule')) == true) {
-      $this->postProcess();
+        return parent::uninstall();
     }
 
-    $this->context->smarty->assign('module_dir', Context::getContext()->link->getBaseLink() . ltrim($this->_path, '/'));
+    public function getContent()
+    {
+        if (((bool)Tools::isSubmit('submitApisearchModule')) == true) {
+            $this->postProcess();
+        }
 
-    $output = $this->context->smarty->fetch($this->local_path . 'views/templates/admin/configure.tpl');
+        $this->context->smarty->assign('module_dir', Context::getContext()->link->getBaseLink() . ltrim($this->_path, '/'));
```

Comment:
For subdir store installation compatibility:

```
$moduleDir = \rtrim(Context::getContext()->link->getBaseLink(), '/').'/modules/'.Defaults::PLUGIN_NAME.'/';
$this->context->smarty->assign('module_dir', $moduleDir);

```

Before: `[base_link_with_subdir]/subdir/module/apisearch/`
After: `[base_link_with_subdir]/module/apisearch/`